Robus/src/msgAlloc.c

```diff
@@ -17,45 +18,134 @@
+     * But this message that we jump is now considered as one of the newest.
+     * To keep slots ordered by time we should move this slot 
      */
+    //******** check if the next slot is available **********
+    if ((alloc_id < MAX_MSG_NB - 1) & (alloc_table[alloc_id + 1].msg_pt == 0))
+    {
+        alloc_id++;
+        return;
+    }
+    if ((alloc_id == MAX_MSG_NB - 1) & (alloc_table[0].msg_pt == 0))
+    {
+        alloc_id = 0;
+        return;
+    }
+    // The next slot is not available we have to find one and slide older slots.
+
+    //******** Find the next available slotslot **********
+    // loop from alloc_id+1 to MAX_MSG_NB to find a free slot
+    //set alloc_id
+    //return if there is one
+    for (uint16_t i = alloc_id + 1; i < MAX_MSG_NB; i++)
+    {
+        if (alloc_table[i].msg_pt == 0)
+        {
+            // we found one
+            // slide older slots to let the next slot available
+            memcpy((void *)&alloc_table[alloc_id + 2], (void *)&alloc_table[alloc_id + 1], (alloc_id + 1 - i) * sizeof(msg_allocator_t));
+            alloc_id++;
+            return;
+        }
+    }
+    // loop from 0 to alloc_id to find a free slot set alloc_id and return if there is one
+    for (uint16_t i = 0; i <= alloc_id; i++)
+    {
+        if (alloc_table[i].msg_pt == 0)
+        {
+            // we found one
+            // slide older slots to let the next slot available
+            // move the begin of the alloc_table
+            memcpy((void *)&alloc_table[1], (void *)&alloc_table[0], i * sizeof(msg_allocator_t));
+            // move the last slot of the alloc_table to the begin of the alloc_table
+            memcpy((void *)&alloc_table[0], (void *)&alloc_table[MAX_MSG_NB - 1], sizeof(msg_allocator_t));
+            // move the end of the alloc_table
+            memcpy((void *)&alloc_table[alloc_id + 2], (void *)&alloc_table[alloc_id + 1], (alloc_id + 1 - MAX_MSG_NB) * sizeof(msg_allocator_t));
+            alloc_id++;
+            return;
+        }
+    }
+    // At this point there is no slot available, put an error somewhere
+    // remove the message on the next slot and make it clean for the module of this slot
+    if (alloc_id < MAX_MSG_NB)
+    {
+        MsgAlloc_ClearSlot(alloc_id + 1);
+        alloc_id++;
+    }
+    else
+    {
+        MsgAlloc_ClearSlot(0);
+        alloc_id = 0;
+    }
 }
 
 /******************************************************************************
  * @brief prepare a buffer space to be usable by cleaning remaining messages and prepare pointers
  * @param from : start of the memory space to clean
  * @param to : start of the memory space to clean
- * @return None
+ * @return error_return_t
  ******************************************************************************/
-static void MsgAlloc_ClearMemSpace(uint8_t *from, uint8_t *to)
+static error_return_t MsgAlloc_ClearMemSpace(void *from, void *to)
```

Comment:
We should replace from and to with a simple size. Based on data_ptr as from we can calculate the to based on the size.